src/Cache/InvalidCacheKeyException.php

```diff
@@ -0,0 +1,7 @@
+<?php
+namespace ScriptFUSION\Porter\Cache;
+
+class InvalidCacheKeyException extends \DomainException implements \Psr\Cache\InvalidArgumentException
```

Comment:
As I understand a DomainException should be thrown in the case that the data doesn't fall within a set of acceptable values. It extends from a LogicException which is thrown if there is an error in the code that can be fixed immediately.

Both the above seem to fit the use case of `generateCacheKey` not generating a PSR-6 compatible string.

RuntimeException fits as we validate at runtime so I am happy to make this change.